libs/repositories/src/UsdRepository/UsdRepositoryCoingecko.ts

```diff
@@ -0,0 +1,116 @@
+import 'reflect-metadata';
+
+import { injectable } from 'inversify';
+import { PricePoint, PriceStrategy, UsdRepository } from './UsdRepository';
+import { COINGECKO_PLATFORMS, coingeckoProClient } from '../coingecko';
+
+/**
+ * Number of days of data to fetch for each price strategy
+ *
+ * Depending on the granularity, we specify the number of days of data to fetch.
+ * Coingecko will auto-choose the granularity based on the number of days. Its actually weird, because if you try to
+ * specify the granularity it will throw you an error (saying that the PRO account is not enough). So its important to
+ * control the granularity by the number of days.
+ *
+ */
+const DAYS_PER_PRICE_STRATEGY: Record<PriceStrategy, number> = {
+  '5m': 1, // 1 day (~288 points)
+  hourly: 5, // 5 Days of hourly data (~120 points)
+  daily: 90, // 90 Days of daily data (~90 points)
+};
+
+@injectable()
+export class UsdRepositoryCoingecko implements UsdRepository {
+  async getUsdPrice(
+    chainId: number,
+    tokenAddress: string
+  ): Promise<number | null> {
```

Comment:
Nitpick/question: since this is part of the interface, the return type needed here explicitly?